python/tk_framework_desktopserver/server_protocol.py

```diff
@@ -434,9 +437,9 @@ def json_reply(self, data):
         :param data: Object Data that will be converted to JSON and sent to client.
         """
         # ensure_ascii allows unicode strings.
-        payload = json.dumps(
-            data, ensure_ascii=False, default=self._json_date_handler,
-        ).encode("utf8")
+        payload = six.ensure_binary(
+            json.dumps(data, ensure_ascii=False, default=self._json_date_handler,)
```

Comment:
Try adding some unicode in your group names for your software entities and you'll get a serialization error here.

You'll need the following:
```
payload = json.dumps(
            data, ensure_ascii=True, default=self._json_date_handler,
        )
```